cores/esp32/IPAddress.cpp

```diff
@@ -413,5 +417,14 @@ IPAddress& IPAddress::from_ip_addr_t(ip_addr_t* addr){
     return *this;
 }
 
+esp_ip6_addr_type_t IPAddress::addr_type(){
```

Comment:
Shouldn't this function then be named `ip6_addr_type()` ?

Reply:
@TD-er we are procrastinating at this point... I like it the way it is. This PR needs to be merged already so we can progress further. Let's not get overly picky?

Reply:
I just mentioned it as it is a "breaking change" whenever it needs to be changed after this has been merged.

Also shouldn't it be a `const` function?
Or does the Espressif code prohibit this (maybe you could use a `const_cast`)

Anyway, do you plan on making a follow-up on this topic after it is merged?
I get it that this is blocking for your other PR with the network rework, so I do see the need for a quick merge.

Reply:
we can revisit everything after the overhaul. Let's get to use it a bit and see what would need adjusting.